### DISCUSSION TITLE
vertical wind (w), potential temperature, output on pressure level

### DISCUSSION THREAD

**Discussion by bnepal22** [ufs-community/ufs-weather-model#2261] [Q&A] [answered]
I am completely new to the UFS model. I am trying to get the vertical wind (w) and potential temperature on a pressure level,  similar to  w100, w200, w500 etc. Is it possible to get it?

**Comment by laurenfrederick** [comment]
Hi @bnepal22, @natalie-perlin and @RatkoVasic-NOAA are taking a look and will get back to you with an answer.

**Comment by RatkoVasic-NOAA** [comment]
@bnepal22 In file diag_table in run directory, you have a list of 3D and 2D fields for output. For your case, there should be already field:
`"gfs_dyn",     "w",           "dzdt",      "fv3_history",    "all",  .false.,  "none",  2`
I looked in control_c192_intel case, and in file atmf012.nc I can find:
```
        float dzdt(time, pfull, grid_yt, grid_xt) ;
                dzdt:_FillValue = 9.99e+20f ;
                dzdt:cell_methods = "time: point" ;
                dzdt:long_name = "vertical wind" ;
                dzdt:missing_value = 9.99e+20f ;
                dzdt:output_file = "atm" ;
                dzdt:units = "m/sec" ;
```
That one is on model levels, here is example of interpolated to standard pressure levels, case control_p8_intel, file GFSPRS.GrbF12:
```
Hera:> wgrib2 -v GFSPRS.GrbF12  | grep DZDT | grep 500
489:31610864:d=2021032206:DZDT Vertical Velocity (Geometric) [m/s]:500 mb:12 hour fcst:
```

**Reply by bnepal22** [reply]
@RatkoVasic-NOAA  Thank you for your response. Yes, I was able to get the 'w' as a 3d variable as suggested by you, however, I was not able to interpolate it to the regular grid. Currently, I am trying to do it for a stretched grid and using freegrid_parallel tool  for to transform the stretched grid to a regular longitude latitude grid for further analysis. Therefore, I wanted to get the w at each pressure level individually as a 2d variable just like zonal (u) and meridional wind (v) by specifying w500, w100 and so on. Currently I am using GFS_v15p2.

**Reply by bnepal22** [reply, marked as answer]
I think I got the solution.  I am now creating a output at model level and then converting hybrid sigma level to pressure level for further analysis using Python. based package PyNgl.

**Comment by RatkoVasic-NOAA** [comment]
@bnepal22 I'm glad that you found solution that is working for you!

**Reply by bnepal22** [reply]
But still, I am confused about, how to identify the list of parameters/ variables that can be output and their corresponding field in which they are registered. In my diag_table I have only a few variables. Where to find the list of variables that can be output? I am currently using  ccpp_suite = "FV3_GFS_v15p2". And running it for S2S scale with grid stretching.

#Current variables I have in my diag_table
"gfs_sfc",   "t2m",           "tmp2m",     "xapp_ins_6h", "all",  .false.,  "none",  2
 "gfs_phys",  "totprcp_ave",   "prate_ave", "xapp_ins_6h", "all",  .false.,  "none",  2
 "gfs_phys",  "u10m",          "ugrd10m",   "xapp_ins_6h", "all",  .false.,  "none",  2
 "gfs_phys",  "v10m",          "vgrd10m",   "xapp_ins_6h", "all",  .false.,  "none",  2
 "gfs_phys",  "tmpmax2m",      "tmax_max2m","xapp_ins_1d","all",  .false.,  "none",  2
 "gfs_phys",  "tmpmin2m",      "tmin_min2m","xapp_ins_1d","all",  .false.,  "none",  2

**Comment by RatkoVasic-NOAA** [comment]
Sure. in this file you have all available diagnostics which you can put in diag_table.
That is place for developers to add new diagnostics for their work.
ufs-weather-model/FV3/ccpp/driver/GFS_diagnostics.F90

**Reply by bnepal22** [reply]
Thank you very much for this information. I will go through it and will do accordingly.
